### PR TITLE
cplugin: allow loading mpv_* symbols dynamically

### PR DESCRIPTION
**Pull request by kasper93**
This enables cplugins on Windows.

Builds on top https://github.com/mpv-player/mpv/pull/7800

Resolves @rossy's concerns from https://github.com/mpv-player/mpv/pull/7800#issuecomment-638776496

### CODE REVIEW

**Comment by Dudemanguy** [comment]
I think you might need the `DEFINE_SYM_PTR` thing for the few functions in `render.h` and `stream_cb.h`.

**Comment by kasper93** [comment]
> I think you might need the DEFINE_SYM_PTR thing for the few functions in render.h and stream_cb.h.

Good catch, added.

File: player/scripting.c [hunk 1]
```diff
@@ -319,8 +398,13 @@ error: ;
 }
 
 const struct mp_scripting mp_scripting_cplugin = {
+    #if _WIN32
```
Comment (Dudemanguy) [inline player/scripting.c:401]:
```suggestion
    #if HAVE_WIN32
```

File: player/scripting.c [hunk 2]
```diff
@@ -295,21 +297,98 @@ bool mp_load_scripts(struct MPContext *mpctx)
 
 #if HAVE_CPLUGINS
 
+#if !_WIN32
```
Comment (Dudemanguy) [inline player/scripting.c:300]:
```suggestion
    #if HAVE_WIN32
```

Comment (Dudemanguy) [inline player/scripting.c:300]:
```suggestion
#if !HAVE_WIN32
```

**Review by Dudemanguy — APPROVED** [review]
No real way for me to test but code looks fine and it doesn't get in the way of linux. Having to remember all of the `init_sym_table` stuff kind of sucks, but it's not like we add things to the libmpv API everyday and I guess there's not really another way to do it.

**Comment by kasper93** [comment]
With different API it could be nicer, but this way we fully preserve compatibility with linux plugins, just need to rebuild with correct flags.

**Pull request merged**